src/vm/hints/blake2s_utils.rs

```diff
@@ -0,0 +1,343 @@
+use std::collections::HashMap;
+
+use num_traits::ToPrimitive;
+
+use super::blake2s_hash::blake2s_compress;
+use super::hint_utils::get_ptr_from_var_name;
+use crate::bigint_u64;
+use crate::serde::deserialize_program::ApTracking;
+use crate::types::relocatable::Relocatable;
+use crate::vm::vm_core::VirtualMachine;
+use crate::{
+    types::relocatable::MaybeRelocatable,
+    vm::{
+        errors::vm_errors::VirtualMachineError,
+        vm_memory::{memory::Memory, memory_segments::MemorySegmentManager},
+    },
+};
+use num_bigint::BigInt;
+use num_traits::FromPrimitive;
+
+fn get_fixed_size_u64_array<const T: usize>(
+    h_range: &Vec<Option<&MaybeRelocatable>>,
+) -> Result<[u64; T], VirtualMachineError> {
+    let mut u64_vec = Vec::<u64>::with_capacity(h_range.len());
+    for element in h_range {
+        let mayberel = element.ok_or(VirtualMachineError::UnexpectMemoryGap)?;
+        let num = if let MaybeRelocatable::Int(num) = mayberel {
+            num
+        } else {
+            return Err(VirtualMachineError::ExpectedInteger(mayberel.clone()));
+        };
```

Comment:
If we write the `.as_int` helper for `MaybeRelocatable` we could do:
```suggestion
        let num = element
            .ok_or(VirtualMachineError::UnexpectMemoryGap)?
            .as_int()?;
```

Reply:
Added helper method `get_int_ref` to MaybeRelocatable